data/basics/userBalance.go

```diff
@@ -180,6 +203,15 @@ func (u AccountData) KeyDilution(proto config.ConsensusParams) uint64 {
 	return proto.DefaultKeyDilution
 }
 
+// IsZero checks if an AccountData value is the same as its zero value.
+func (ad AccountData) IsZero() bool {
+	if ad.Currencies != nil && len(ad.Currencies) == 0 {
+		ad.Currencies = nil
+	}
+
+	return reflect.DeepEqual(ad, AccountData{})
```

Comment:
Can we use a non-reflect way to do that ? It doesn't look like it would too complicated to consider other solutions here.

Reply:
The main benefit of reflect that I was going for was that it makes it impossible to miss checking a field.

I agree it doesn't really require reflection; just checking that every field is zero.